### PR TITLE
Fixes for issues discovered while running one oracle

### LINKED ISSUE #121: Consequent publish data point operation return "double spending" node error
Discovered in #61

After the first successful publish datapoint node request, the second publish request (in 30 sec) end up with "double spending" node error. 
The reason for the second request is that in 30 secs there is no published box in wallet scans, thus publishing again.

There are the following options on how to fix this behavior:
- ignore this particular node error, eventually, (a few minutes) wallet scan will return the published box;
- save in-memory status that datapoint was sent to the node in this epoch;

### LINKED ISSUE #127: RefreshAction failed with error: refresh box not found
Discovered in #61

### LINKED ISSUE #128: Refresh action with not enough data point throws fatal error
Discovered in #61

The refresh action should gracefully handle not enough datapoints logging an error and continue with the main loop.

### PR DESCRIPTION
**Pull request by greenhat**
- Close #121
- Close #127
- Close #128

### CODE REVIEW

File: core/src/serde.rs [hunk 1]
```diff
@@ -190,9 +190,9 @@ impl TryFrom<OracleConfigSerde> for OracleConfig {
 
         let refresh_box_wrapper_inputs = RefreshBoxWrapperInputs::checked_load(
             refresh_contract_parameters.clone(),
-            c.token_ids.refresh_nft_token_id.clone(),
             c.token_ids.oracle_token_id.clone(),
-            c.token_ids.reward_token_id.clone(),
+            c.token_ids.pool_nft_token_id.clone(),
+            c.token_ids.refresh_nft_token_id.clone(),
```
Comment (greenhat) [inline core/src/serde.rs:195]: This is exactly the kind of error I wanted to eliminate when I made #47

File: core/src/main.rs [hunk 2]
```diff
@@ -320,8 +327,9 @@ fn handle_oracle_command(command: Command) {
 }
 
 fn main_loop_iteration(op: &OraclePool, read_only: bool) -> std::result::Result<(), anyhow::Error> {
-    let height = current_block_height()?;
+    let height = current_block_height()? as u32;
     let wallet = WalletData::new();
+    let networt_change_address = get_change_address_from_node()?;
```
Comment (kettlebell) [inline core/src/main.rs:332]: Just a small typo. It should read `network_change_address`.

File: core/src/oracle_config.rs [hunk 3]
```diff
@@ -107,9 +107,9 @@ impl OracleConfig {
         )?;
         let refresh_box_wrapper_inputs = RefreshBoxWrapperInputs::build_with(
             bootstrap.refresh_contract_parameters.clone(),
-            token_ids.refresh_nft_token_id.clone(),
             token_ids.oracle_token_id.clone(),
-            token_ids.reward_token_id.clone(),
+            token_ids.pool_nft_token_id.clone(),
```
Comment (kettlebell) [inline core/src/oracle_config.rs:111]: Damn, I think I should have picked up on this in a prior review. As you say, it's why we need #47.

File: core/src/oracle_state.rs [hunk 4]
```diff
@@ -511,46 +487,34 @@ fn register_and_save_scans_inner() -> std::result::Result<(), Error> {
         OracleContract::checked_load(&config.oracle_box_wrapper_inputs.contract_inputs)?
             .ergo_tree();
 
-    let mut scans = vec![
```
Comment (kettlebell) [inline core/src/oracle_state.rs:514]: Nice refactoring here!

**Review by kettlebell — CHANGES_REQUESTED** [review]
Looks great! Just a small typo to fix.

File: core/src/main.rs (hunk 2 above)
Comment (greenhat) [reply core/src/main.rs]: Thank you! Fixed.

File: core/src/oracle_config.rs (hunk 3 above)
Comment (greenhat) [reply core/src/oracle_config.rs:111]: Yes, #47 cannot come soon enough.

File: core/src/oracle_state.rs (hunk 4 above)
Comment (greenhat) [reply core/src/oracle_state.rs:514]: Thanks!

**Review by kettlebell — APPROVED** [review]
Great thanks!

**Pull request merged**